Sources/Brave/Frontend/Shields/SubmitReportView.swift

```diff
@@ -0,0 +1,143 @@
+  @Environment(\.dismiss) private var dismiss: DismissAction
+  let url: URL
+  let isPrivateBrowsing: Bool
+  
+  @State private var additionalDetails = ""
+  @State private var contactDetails = ""
+  @State private var isSubmittingReport = false
+  @State private var isSubmitted = false
+  
+  private var scrollContent: some View {
+    ScrollView {
+      VStack(alignment: .leading, spacing: 16) {
+        Text(Strings.Shields.reportBrokenSiteBody1)
+        Text(url.absoluteString)
+          .foregroundStyle(Color(braveSystemName: .textInteractive))
+        Text(Strings.Shields.reportBrokenSiteBody2)
+          .font(.footnote)
+        BraveTextEditor(
+          text: $additionalDetails,
+          prompt: Strings.Shields.reportBrokenAdditionalDetails
+        ).frame(height: 80)
+        
+        VStack(alignment: .leading, spacing: 4) {
+          Text(Strings.Shields.reportBrokenContactMe).font(.caption)
+          TextField(
+            Strings.Shields.reportBrokenContactMe,
+            text: $contactDetails, prompt: Text(Strings.Shields.reportBrokenContactMeSuggestions)
+          )
+          .textContentType(.emailAddress)
+          .textFieldStyle(BraveTextFieldStyle())
+          .autocorrectionDisabled()
+          .textInputAutocapitalization(.never)
+        }
+      }
+      .padding()
+      .foregroundStyle(Color(braveSystemName: .textSecondary))
+    }
+    .background(Color(.braveBackground))
+    .foregroundStyle(Color(braveSystemName: .textSecondary))
+    .navigationTitle(Strings.Shields.reportABrokenSite)
+    .toolbar {
+      ToolbarItem(placement: .cancellationAction) {
+        Button(Strings.cancelButtonTitle) {
+          dismiss()
+        }
+        .disabled(isSubmittingReport)
+        .hidden(isHidden: isSubmitted)
+        .animation(.smooth.speed(0.2), value: isSubmitted)
+      }
+      
+      ToolbarItem(placement: .confirmationAction) {
+        if isSubmitted {
+          Button(Strings.close, action: {
+            dismiss()
+          })
+        } else {
+          Button(Strings.Shields.reportBrokenSubmitButtonTitle, action: {
+            withAnimation {
+              isSubmittingReport = true
+            }
+            
+            Task { @MainActor in
+              await createAndSubmitReport()
+              
+              withAnimation {
+                isSubmitted = true
+              }
+              
+              try await Task.sleep(seconds: 4)
+              dismiss()
+            }
+          }).disabled(isSubmittingReport)
+        }
+      }
+    }
+    .overlay {
+      if isSubmittingReport {
+        ProgressView()
+          .progressViewStyle(.braveCircular(size: .normal, tint: .braveBlurpleTint))
+          .frame(maxWidth: .infinity, maxHeight: .infinity)
+          .background(Color(.braveBackground).opacity(0.5).ignoresSafeArea())
+          .animation(.easeInOut, value: isSubmittingReport)
+      }
+    }
+    .overlay {
+      if isSubmitted {
```

Comment:
Why not just push this onto the stack instead? (+ `navigationBarBackButtonHidden` on it)

Reply:
> Why not just push this onto the stack instead? (+ `navigationBarBackButtonHidden` on it)
seems like a lot more code and not as nice a result, no?

Reply:
From an accessibility perspective this is more confusing: No screen change notification, the VoiceOver is also going to read the contents underneath still despite the elements on top being there since you didn't actually hide them, just overlaid things.

Reply:
ahh good point. But if I push it to the stack, then I guess i'll have to do it on the view controller side?

Reply:
Just use a `NavigationLink` that takes an `isActive` binding, place it in the view hierarchy (in a background modifier with an `EmptyView` label), update the binding when it you finish submitting

Reply:
@kylehickinson This is a deprecated API. But ok, once we drop iOS 15 we can move to NavigationStack with navigationDestination I believe, so I went with this approach